FE restart failed due to Materialized View

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)

FE cannot restart when replaying the journal, which contains inactive MV.


```java
2023-08-26 12:07:26,524 WARN (stateChangeExecutor|91) [GlobalStateMgr.replayJournal():1883] got interrupt exception or inconsistent exception when replay 
journal 3070926, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10097
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:954) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1930) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1881) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToLeader(GlobalStateMgr.java:1050) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$100(GlobalStateMgr.java:298) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$1.transferToLeader(GlobalStateMgr.java:655) ~[starrocks-fe.jar:?]
        at com.starrocks.ha.StateChangeExecutor.runOneCycle(StateChangeExecutor.java:86) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) ~[starrocks-fe.jar:?]
Caused by: com.starrocks.sql.analyzer.SemanticException: Create materialized view from inactive materialized view: cyb_dwd_catarc_mix_merge_mon_v
        at com.starrocks.sql.analyzer.MaterializedViewAnalyzer.lambda$getBaseTableInfos$0(MaterializedViewAnalyzer.java:114) ~[starrocks-fe.jar:?]
        at java.util.HashMap.forEach(HashMap.java:1288) ~[?:1.8.0_131]
        at com.starrocks.sql.analyzer.MaterializedViewAnalyzer.getBaseTableInfos(MaterializedViewAnalyzer.java:106) ~[starrocks-fe.jar:?]
        at com.starrocks.alter.Alter.processChangeMaterializedViewStatus(Alter.java:336) ~[starrocks-fe.jar:?]
        at com.starrocks.alter.Alter.replayAlterMaterializedViewStatus(Alter.java:574) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayAlterMaterializedViewStatus(GlobalStateMgr.java:3099) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:294) ~[starrocks-fe.jar:?]
        ... 7 more
```


### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
- 2.5.*, 3.0.*, 3.1.*

--- a/fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java
@@ -297,9 +297,15 @@ public class AlterJobMgr {
         long tableId = log.getTableId();
         Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
         db.writeLock();
+        MaterializedView mv = null;
         try {
-            MaterializedView mv = (MaterializedView) db.getTable(tableId);
+            mv = (MaterializedView) db.getTable(tableId);
             alterMaterializedViewStatus(mv, log.getStatus(), true);
+        } catch (Throwable e) {
+            if (mv != null) {
+                LOG.warn("replay alter materialized-view status failed: {}", mv.getName(), e);
+                mv.setInactiveAndReason("replay alter status failed: " + e.getMessage());
+            }
         } finally {
             db.writeUnlock();
         }
@@ -310,13 +316,25 @@ public class AlterJobMgr {
         long materializedViewId = log.getId();
         String newMaterializedViewName = log.getNewMaterializedViewName();
         Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
-        MaterializedView oldMaterializedView = (MaterializedView) db.getTable(materializedViewId);
-        db.dropTable(oldMaterializedView.getName());
-        oldMaterializedView.setName(newMaterializedViewName);
-        db.registerTableUnlocked(oldMaterializedView);
-        updateTaskDefinition(oldMaterializedView);
-        LOG.info("Replay rename materialized view [{}] to {}, id: {}", oldMaterializedView.getName(),
-                newMaterializedViewName, oldMaterializedView.getId());
+
+        db.writeLock();
+        MaterializedView oldMaterializedView = null;
+        try {
+            oldMaterializedView = (MaterializedView) db.getTable(materializedViewId);
+            db.dropTable(oldMaterializedView.getName());
+            oldMaterializedView.setName(newMaterializedViewName);
+            db.registerTableUnlocked(oldMaterializedView);
+            updateTaskDefinition(oldMaterializedView);
+            LOG.info("Replay rename materialized view [{}] to {}, id: {}", oldMaterializedView.getName(),
+                    newMaterializedViewName, oldMaterializedView.getId());
+        } catch (Throwable e) {
+            if (oldMaterializedView != null) {
+                oldMaterializedView.setInactiveAndReason("replay rename failed: " + e.getMessage());
+                LOG.warn("replay rename materialized-view failed: {}", oldMaterializedView.getName(), e);
+            }
+        } finally {
+            db.writeUnlock();
+        }
     }
 
     private void updateTaskDefinition(MaterializedView materializedView) {
@@ -337,8 +355,8 @@ public class AlterJobMgr {
             return;
         }
         db.writeLock();
+        MaterializedView oldMaterializedView = null;
         try {
-            MaterializedView oldMaterializedView;
             final MaterializedView.MvRefreshScheme newMvRefreshScheme = new MaterializedView.MvRefreshScheme();
 
             oldMaterializedView = (MaterializedView) db.getTable(id);
@@ -365,6 +383,12 @@ public class AlterJobMgr {
                     oldMaterializedView.getName(), refreshType.name(), asyncRefreshContext.getStartTime(),
                     asyncRefreshContext.getStep(),
                     asyncRefreshContext.getTimeUnit(), oldMaterializedView.getId());
+        } catch (Throwable e) {
+            if (oldMaterializedView != null) {
+                oldMaterializedView.setInactiveAndReason("replay failed: " + e.getMessage());
+                LOG.warn("replay change materialized-view refresh scheme failed: {}",
+                        oldMaterializedView.getName(), e);
+            }
         } finally {
             db.writeUnlock();
         }
@@ -377,8 +401,9 @@ public class AlterJobMgr {
 
         Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
         db.writeLock();
+        MaterializedView mv = null;
         try {
-            MaterializedView mv = (MaterializedView) db.getTable(tableId);
+            mv = (MaterializedView) db.getTable(tableId);
             TableProperty tableProperty = mv.getTableProperty();
             if (tableProperty == null) {
                 tableProperty = new TableProperty(properties);
@@ -386,6 +411,11 @@ public class AlterJobMgr {
             } else {
                 tableProperty.modifyTableProperties(properties);
                 tableProperty.buildProperty(opCode);
+            }
+        } catch (Throwable e) {
+            if (mv != null) {
+                mv.setInactiveAndReason("replay failed: " + e.getMessage());
+                LOG.warn("replay alter materialized-view properties failed: {}", mv.getName(), e);
             }
         } finally {
             db.writeUnlock();

